lib/FrameManager.js

```diff
@@ -282,16 +305,32 @@ class Frame {
       this._parentFrame._childFrames.add(this);
   }
 
+  _navigationScheduled() {
+    if (!this._defaultContext)
```

Comment:
this misses the case when:
- frame schedules navigation
- a runtime context is coming afterwards

In this case, the `_navigationScheduled` doesn't have any effect.